pkg/auth/event_policy.go

```diff
@@ -210,17 +216,18 @@ func resolveSubjectsFromReference(resolver *resolver.AuthenticatableResolver, re
 	return objFullSANames, nil
 }
 
-// SubjectContained checks if the given sub is contained in the list of allowedSubs
+// SubjectAndFiltersPass checks if the given sub is contained in the list of allowedSubs
 // or if it matches a prefix pattern in subs (e.g. system:serviceaccounts:my-ns:*)
-func SubjectContained(sub string, allowedSubs []string) bool {
-	for _, s := range allowedSubs {
-		if strings.EqualFold(s, sub) {
-			return true
-		}
+func SubjectAndFiltersPass(ctx context.Context, sub string, allowedSubsWithFilters []filtersBySubjects, event *cloudevents.Event, logger *zap.SugaredLogger) bool {
```

Comment:
Not sure, what I am missing here, but isn't the API for filters for the whole EventPolicy and not aligned to some subjects instead of like here "filters by subjects"?

Reply:
Ah ok. According to https://github.com/knative/eventing/pull/8122/files#diff-5a45b6a786dbd89c11dc49f5d111d8b06700e9770b13a1f37d4896c3625f5a20R148 it seems to be more a "filters and subjects"

Reply:
Yeah, the way I was thinking about it is if the subject matches the `.from` subjects, then the event is matched against the filters for that policy. I agree the name isn't great here, I was having a hard time finding a good one